screwdriver/screwdriver.go

```diff
@@ -44,70 +43,42 @@ type Step struct {
 	Command string `json:"command"`
 }
 
-// EnvVar is an environment slice
-type EnvVar []struct {
-	Key   string
-	Value string
-}
-
-// UnmarshalJSON replaces JSON of a normal associative array to EnvVar
-func (en *EnvVar) UnmarshalJSON(data []byte) error {
-	for _, pair := range regexpForUnmarshal.FindAllStringSubmatch(string(data), -1) {
-		*en = append(*en, struct {
-			Key   string
-			Value string
-		}{
-			pair[1],
-			pair[2],
-		})
-	}
-	return nil
-}
+// EnvVars is an environment slice
+type EnvVars []map[string]string
 
-// MarshalJSON replaces EnvVar to JSON of a normal associative array
-func (en EnvVar) MarshalJSON() ([]byte, error) {
-	outputArray := make([]string, len(en))
-	for i, pair := range en {
-		outputArray[i] = "{\"" + pair.Key + "\":\"" + pair.Value + "\"}"
+// UnmarshalJSON replaces JSON of a normal associative array to EnvVars
+func (en *EnvVars) UnmarshalJSON(data []byte) error {
+	inputbytes := []byte(data)
+	orderedMap := ordered.NewOrderedMap()
+	err := json.Unmarshal(inputbytes, orderedMap)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal: %v", err)
 	}
-	output := "[" + strings.Join(outputArray, ",") + "]"
-	return []byte(output), nil
-}
 
-// Get the newest Value whose Key is key
-func (en EnvVar) Get(key string) string {
-	s := ""
-	for _, e := range en {
-		if e.Key == key {
-			s = e.Value
+	iter := orderedMap.EntriesIter()
+	for {
+		pair, ok := iter()
+		if !ok {
+			break
 		}
+		*en = append(*en, map[string]string{pair.Key: pair.Value.(string)})
 	}
-	return s
-}
 
-// Set adds (key, val)
-func (en *EnvVar) Set(key string, val string) {
-	*en = append(*en, struct {
-		Key   string
-		Value string
-	}{
-		key,
-		val,
-	})
+	return nil
 }
 
-// Merge en2 to en
-func (en *EnvVar) Merge(en2 EnvVar) {
-	for _, e := range en2 {
-		en.Set(e.Key, e.Value)
+// Merge associative array to EnvVars
+func (en *EnvVars) Merge(en2 map[string]string) {
```

Comment:
This function may be better named `AppendAll` than `Merge`

The name merge is reminiscent of the following behavior:
```
o1 = { a: aaa, b: bbb}
o2 = { b: foo, c: bar}

o3 = Merge(a, b)
// o3: { a: aaa, b: foo, c: bar}
```

Reply:
I fixed.